pkg/aws/manager_ec2_security_group_test.go

```diff
@@ -0,0 +1,196 @@
+		logger        *logrus.Entry
+		Ec2Api        ec2iface.EC2API
+	}
+	type args struct {
+		clusterId string
+		tags      map[string]string
+		dryRun    bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []*clusterservice.ReportItem
+		wantErr string
+	}{
+		{
+			name: "successful run",
+			fields: fields{
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.deleteSecurityGroupFn = func(input *ec2.DeleteSecurityGroupInput) (*ec2.DeleteSecurityGroupOutput, error) {
+						return &ec2.DeleteSecurityGroupOutput{}, nil
+					}
+				}),
+				taggingClient: func() *taggingClientMock {
+					client, err := fakeTaggingClient(func(c *taggingClientMock) error {
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return client
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				tags:      map[string]string{},
+				dryRun:    false,
+			},
+			want: []*clusterservice.ReportItem{
+				fakeReportItemDeleting(),
+			},
+		}, {
+			name: "fail when getting resources via tags returns an error",
+			fields: fields{
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.deleteSecurityGroupFn = func(input *ec2.DeleteSecurityGroupInput) (*ec2.DeleteSecurityGroupOutput, error) {
+						return &ec2.DeleteSecurityGroupOutput{}, nil
+					}
+				}),
+				taggingClient: func() *taggingClientMock {
+					client, err := fakeTaggingClient(func(c *taggingClientMock) error {
+						c.GetResourcesFunc = func(in1 *resourcegroupstaggingapi.GetResourcesInput) (output *resourcegroupstaggingapi.GetResourcesOutput, e error) {
+							return nil, errors.New("some tagging error")
+						}
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return client
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				tags:      map[string]string{},
+				dryRun:    false,
+			},
+			wantErr: "error getting security group to delete: failed to filter vpc: some tagging error",
+		},
+		{
+			name: "fail when vpc deletion returns an error",
+			fields: fields{
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.deleteSecurityGroupFn = func(input *ec2.DeleteSecurityGroupInput) (*ec2.DeleteSecurityGroupOutput, error) {
+						return nil, errors.New("some error deleting subnet")
+					}
+				}),
+				taggingClient: func() *taggingClientMock {
+					client, err := fakeTaggingClient(func(c *taggingClientMock) error {
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return client
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				tags:      map[string]string{},
+				dryRun:    false,
+			},
+			wantErr: "failed to delete security group: some error deleting subnet",
+		},
+		{
+			name: "succeeds with status in progress if dry run is false",
```

Comment:
Is this the same as the successful test case?

Reply:
:+1: